Safety/Inc/PWM.hpp

```diff
@@ -54,4 +73,57 @@ class PWMChannel {
 	uint32_t pwmPeriod = 20000;
 	uint32_t min_signal = 950; //standard for 50hz pwm signal, with a 100 Hz margin added to ensure we can reach the extreme values.
 	uint32_t max_signal = 2050;
+
+	/**
+  * @brief Prepares the Dshot data frame including the data, telemetry bit, and checksum
+  * @param throttlePercentage Trottle percentage output from the PID loop, 0-100%
+  * @param telemetry Bool indicating whether to set the telemetry bit or not
+  * @retval 16 bit dshot data frame
+  */    
+    uint16_t dshotPrepareFrame(uint8_t throttlePercentage, bool telemetry);
```

Comment:
should we be looking to change throttlePercentage to a float now that we have significantly more resolution with DSHOT? Maybe not for this PR, but something to consider

Reply:
That makes sense, I think with DSHOT we can definitely handle the increased sensitivity. I guess just a matter if it would actually make a difference or not...